services/blockstorage/service.go

```diff
@@ -74,19 +77,32 @@ func (s *service) CommitBlock(input *services.CommitBlockInput) (*services.Commi
 		return nil, err
 	}
 
-	s.updateLastCommittedBlockHeightAndTimestamp(txBlockHeader)
+	s.updateLastCommittedBlock(input.BlockPair)
 
 	s.reporting.Info("Committed a block", instrumentation.BlockHeight(txBlockHeader.BlockHeight()))
 
 	return nil, nil
 }
 
-func (s *service) updateLastCommittedBlockHeightAndTimestamp(txBlockHeader *protocol.TransactionsBlockHeader) {
+func (s *service) updateLastCommittedBlock(block *protocol.BlockPairContainer) {
 	s.lastBlockLock.Lock()
 	defer s.lastBlockLock.Unlock()
 
-	s.lastCommittedBlockHeight = txBlockHeader.BlockHeight()
-	s.lastCommittedBlockTimestamp = txBlockHeader.Timestamp()
+	s.lastCommittedBlock = block
+}
+
+func (s *service) lastCommittedBlockHeight() primitives.BlockHeight {
+	if s.lastCommittedBlock == nil {
+		return 0
+	}
+	return s.lastCommittedBlock.TransactionsBlock.Header.BlockHeight()
```

Comment:
You only protect under mutex the write without protecting the read. Is this our convention? Are we 100% sure that this is safe? isn't it an implementation detail of how the code is compiled?

If this is indeed our convention, we should communicate to everybody because lots of the code in the system is locking for both read and write.

If it's not our convention, we should fix,

Reply:
its not about convention but rather what is required. when we were looking at the code figuring out the spots requiring protection we did not find any place requiring read protection (we actually did, but refactored to avoid the lock)
by the way, we since refactored again and i am not sure this code even needs write protection, but its irrelevant to discuss this in detail as we will refactor more until we get to persistence